src/javascript/app/pages/trade/barriers.js

```diff
@@ -160,14 +161,75 @@ const Barriers = (() => {
         Defaults.remove('barrier', 'barrier_high', 'barrier_low');
     };
 
-    const validateBarrier = () => {
+    const showError = (barrier) => {
+        if (!barrier.classList.contains('error-field')) {
+            barrier.classList.add('error-field');
+        }
+        const error_node = barrier.parentNode.lastElementChild.firstElementChild;
+        if (error_node.classList.contains('invisible')) {
+            error_node.classList.remove('invisible');
+        }
+    };
+
+    const resolveError = (barrier) => {
+        if (barrier.classList.contains('error-field')) {
+            barrier.classList.remove('error-field');
+        }
+        const error_node = barrier.parentNode.lastElementChild.firstElementChild;
+        if (!error_node.classList.contains('invisible')) {
+            error_node.classList.add('invisible');
+        }
+    };
+
+    /**
+    * Show an error on the target Barrier.
+    *
+    * @param {Object} barrier_1   the target barrier object for prompting error
+    * @param {Object} barrier_2   barrier object whose errors will be resolved
+    */
+    const showThisError = (barrier_1, barrier_2) => {
```

Comment:
we don't normally use `This` in names, probably better to be more specific about what `This` is.